concrete/src/Authentication/Type/OAuth/GenericOauthTypeController.php

```diff
@@ -263,6 +263,13 @@ protected function createUser()
         }
 
         $email = $this->getEmail();
+
+        // Sometimes we may not receive email address from SNS apps
+        // Lets create a dummy email address if it's because it's already passed all validations
+        if (!$email) {
+            $email = $this->getUniqueId() . '@' . $this->getHandle();
```

Comment:
This is not a good solution. We need to ask for their email the same way we do with Twitter which just shows an additional interstitial if the email isn't provided by the oauth service.

Reply:
Fixed on https://github.com/concrete5/concrete5/pull/6018/commits/f162516be17421c2a0a5804868c63f839ec27f80